### PR TITLE
[Clock] Add `ClockAwareTrait` to help write time-sensitive classes

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Resolves https://github.com/symfony/symfony-docs/issues/17506

### CODE REVIEW

File: components/clock.rst [hunk 1]
```diff
@@ -104,3 +104,48 @@ the system up to nanosecond precision. It can be used to measure the elapsed
 time between two calls without being affected by inconsistencies sometimes introduced
 by the system clock, e.g. by updating it. Instead, it consistently increases time,
 making it especially useful for measuring performance.
+
+Using a Clock inside a Service
+------------------------------
+
+If your application uses :ref:`service autoconfiguration <services-autoconfigure>`,
+any service whose class uses :class:`Symfony\\Component\\Clock\\ClockAwareTrait` trait will
```
Comment (OskarStark) [inline components/clock.rst]:
```suggestion
any service whose class uses the :class:`Symfony\\Component\\Clock\\ClockAwareTrait` will
```

File: components/clock.rst [hunk 2]
```diff
@@ -104,3 +104,48 @@ the system up to nanosecond precision. It can be used to measure the elapsed
...
+passed as a service. Services using this trait will also benefit of a ``now()`` method.
+
+By using this trait in your services to retrieve the current time, it will be easier to
+write time-sensitive classes. For example, by using the ``MockClock`` implementation as the
+default one during tests, you will have full control of the "current time".
+
+This is also true for the other clock implementations, as you will be assured to always use
+the same clock implementation across your code each time you need it.
+
+The trait can be used like this::
```
Comment (OskarStark) [inline components/clock.rst]:
```suggestion
the same clock implementation across your code each time you need it::
```
?

File: components/clock.rst [hunk 3]
```diff
@@ -104,3 +104,48 @@ the system up to nanosecond precision. It can be used to measure the elapsed
...
+    }
+
+Thanks to the ``ClockAwareTrait`` and by using the ``MockClock`` implementation,
+you will be able to test different times returned by ``now()``. This allows you
+to test every case of your method without the need of actually being in a month
+or another.
+
+.. versionadded:: 6.3
+
+    The ``ClockAwareTrait`` was introduced in Symfony 6.3.
```
Comment (OskarStark) [inline components/clock.rst]:
```suggestion
    The :class:`Symfony\\Component\\Clock\\ClockAwareTrait` was introduced in Symfony 6.3.
```

**Comment by alexandre-daubois** [comment]
That's all updated. Thank you Oskar!

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Alex for yet another nice contribution!

While merging I reworded it a bit to change the order. The original order was "how to use this" + "why you should use this" ... and I think the reverse order makes more sense for the reader: "why" + "how". Thanks.

**Pull request merged**